packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -0,0 +1,489 @@
+        amount1Min: 0,
+        recipient: depositor.address,
+        deadline: Math.floor(await getNow(ethers.provider) + 8640000),// uint256
+      }
+
+      await weth.connect(depositor).deposit({value: collateralToLp})
+      await weth.connect(depositor).approve(positionManager.address, ethers.constants.MaxUint256)
+      await wSqueeth.connect(depositor).approve(positionManager.address, ethers.constants.MaxUint256)  
+      await (positionManager as INonfungiblePositionManager).connect(depositor).mint(mintParam)
+    })
+
+    before("open short amount more than amount in LP position" , async () => {
+      const normFactor = await controller.getExpectedNormalizationFactor()
+      const mintWSqueethAmount = ethers.utils.parseUnits('35')
+      const mintRSqueethAmount = mintWSqueethAmount.mul(normFactor).div(one)
+      const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+      const collateralAmount = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      const tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+      const tokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+
+      await (positionManager as INonfungiblePositionManager).connect(depositor).approve(controller.address, tokenId)
+      await controller.connect(depositor).mintWPowerPerpAmount(0, mintWSqueethAmount, tokenId)
+    })
+
+    it("close LP position deposited as collateral in vault", async () => {
+      const vaultId = (await shortSqueeth.nextId()).sub(1);
+      const uniTokenId =  (await controller.vaults(vaultId)).NftCollateralId;
+      const vaultBefore = await controller.vaults(vaultId); 
+      const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = vaultBefore.shortAmount.mul(scaledEthPrice).div(one)
+      const collateralToFlashloan = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 420, true)
+      const slippage = BigNumber.from(3).mul(BigNumber.from(10).pow(16))
+      const limitPriceEthPerPowerPerp = squeethPrice.mul(one.add(slippage)).div(one);
+      const positionBefore = await (positionManager as INonfungiblePositionManager).positions(uniTokenId);
+      const flashloanCloseVaultLpNftParam = {
+        vaultId: vaultId,
+        tokenId: uniTokenId,
+        liquidity: positionBefore.liquidity,
+        liquidityPercentage: ethers.utils.parseUnits('1'),
+        wPowerPerpAmountToBurn: vaultBefore.shortAmount.toString(),
+        collateralToFlashloan: collateralToFlashloan.toString(),
+        collateralToWithdraw: collateralToFlashloan.toString(),
```

Comment:
not sure why we withdraw this

collateralToWithdraw is in addition to the flashloaned amt (so maybe this should be 0, to leave collateral in vault or should be vault.collateralAmount)

Reply:
fixed